climada/hazard/storm_europe.py

```diff
@@ -199,6 +206,209 @@ def _read_one_nc(self, file_name, centroids):
+                stacked = xr.concat([stacked,ds_i], 'valid_time')
+
+        # create intensity matrix with max for each full day
+        if not (run_date.hour == 0 or run_date.hour == 12):
+            LOGGER.warning('The event definition is inaccuratly implemented '+
+                           'for starting times, which are not 00H or 12H.')
+
+        stacked = stacked.assign_coords(date=('valid_time',stacked["valid_time"].dt.floor("D")))
+        if event_date:
+            stacked2 = stacked.sel(valid_time=event_date.strftime('%Y-%m-%d')).groupby('date').max()
+            considered_dates = np.datetime64(event_date)
+        else:
+            time_covered_step = stacked['valid_time'].diff('valid_time')
+            time_covered_day = time_covered_step.groupby('date').sum()
+            # forecast run should cover at least 18 hours of a day
+            days_to_consider = time_covered_day > np.timedelta64(18,'h') 
+            stacked2 = stacked.groupby('date').max().sel(date=days_to_consider)
+            unique_dates = stacked['valid_time'].groupby('date').min()
+            considered_dates = unique_dates.sel(date=days_to_consider).values
+        stacked2 = stacked2.stack(event=('date', 'number'))
+        stacked2 = stacked2.where(stacked2 > self.intensity_thres)
+        stacked2 = stacked2.fillna(0)
+
+
+        # fill in values from netCDF
+        self.intensity = sparse.csr_matrix(stacked2.gust.T)
+        self.event_id = np.arange(stacked2.date.size)+1
+
+        # fill in default values
+        self.units = 'm/s'
+        self.fraction = self.intensity.copy().tocsr()
+        self.fraction.data.fill(1)
+        self.orig = np.ones_like(self.event_id)*False
+        self.orig[(stacked2.number == 1).values] = True
+
+        self.date = np.repeat(
+            np.array(datetime64_to_ordinal(considered_dates)),
+            stacked.number.size
+            )
+        self.event_name = [date_i + '_ens' + str(ens_i)
+                           for date_i, ens_i in zip(date_to_str(self.date),
+                                                    stacked2.number.values)
+                           ]
+        self.frequency = np.divide(
+                np.ones_like(self.event_id),
+                stacked.number.size)
+        if not description:
+            description = ('icon weather forecast windfield ' +
+                           'for run startet at ' +
+                           run_date.strftime('%Y%m%d%H'))
+
+        self.tag = TagHazard(
+            HAZ_TYPE, 'Hazard set not saved, too large to pickle',
+            description=description
+            )
+        self.check()
+
+
+
+        # delete generated .grib2 and .4cc40.idx files
+        for ind_i, file_i in enumerate(file_names):
+            gripfile_path_i = os.path.join(file_i[:-4])
+            os.remove(gripfile_path_i)
+            idxfile_path_i = gripfile_path_i + '.4cc40.idx'
+            os.remove(idxfile_path_i)
+
+        if delete_raw_data:
```

Comment:
does the downloaded `icon_grid_0028_R02B07_N02.nc` count as raw data that should be deleted?

Reply:
No, my reasoning was: as the daily forecasts, will change everyday and `delete_raw_data` basically gives a choice if these daily forecasts should be archived. The `icon_grid...` file on the other hand is used in every call of this function and is saved permanently for later reuse.